src/compiler/checker.ts

```diff
@@ -256,6 +256,20 @@ namespace ts {
         VoidIsNonOptional = 1 << 1,
     }
 
+    const enum IntrinsicTypeKind {
+        Uppercase,
+        Lowercase,
+        Capitalize,
+        Uncapitalize
+    }
+
+    const intrinsicTypeKinds: ReadonlyESMap<string, IntrinsicTypeKind> = new Map(getEntries({

```

Comment:
Maybe also change `ThisType` into an intrinsic?

Reply:
Why? `ThisType` is just a marker type that doesn't have any effect on its type argument. No reason to change it.

Reply:
I thought "intrinsic" means "hey I'm compiler magic" and ThisType is a compiler magic too 👀

Reply:
There are several "magic" types already that aren't intrinsic. For example `Object`, `Function`, `Array<T>`, `Promise<T>`, `Iterator<T>`, and others all receive special treatment in one way or another. The particular role "intrinsic" plays is to indicate that the _implementation_ of the type is provided by the compiler. In the case of `ThisType`, there really is nothing interesting about the implementation because `ThisType<T>` is the same as `T`. All we do with `ThisType` is to take special hints when a reference to it occurs in a contextual type, but otherwise there's nothing special about it.